### DISCUSSION TITLE
Proof of concept of Taskchampion sync client implemented in accessible language other than Rust, e.g. Python

### DISCUSSION THREAD

**Discussion by bradyt** [GothenburgBitFactory/taskwarrior#2967] [General]
I found it invaluable to refer to the [taskd-client-py](https://github.com/jrabbit/taskd-client-py) project to get a sense of how to write a taskd client, for Taskserver.

I've noticed the project may be migrating from Taskserver, to a server implemented in Rust, via Taskchampion efforts. This can be a little discouraging when working on third party clients, for example there is more than one mobile client implementing a taskd client.

I am hoping someone, maybe even within the GBF team, will write a second minimal client to this new Taskchampion in another language, easily accessible to hobbyists, for example, in Python which I think a lot of people are familiar with, and find it easy enough to read and study.

Such a minimal proof-of-concept client may serve as a complement to any existing specs for the new Taskchampion server sync mechanism.

**Comment by djmitche** [comment]
While it is possible to duplicate this functionality in another language, the intent is to have language bindings so that the TC client code can be used from Python (or Perl or Go or what-have-you) without requiring a re-implementation.

In fact, this is one of the reasons to have made this change: to enable more applications in more languages to participate in TW, without embedding a `task` binary and using its CLI, and without requiring third-party implementations of the sync protocol.

Are you able to help out in any way?  Perhaps by working on a Python binding for taskchampion?
